src/audio/dai.c

```diff
@@ -222,6 +222,9 @@ static void dai_free(struct comp_dev *dev)
 		dd->chan->dev_data = NULL;
 	}
 
+	if (dev->state != COMP_STATE_PREPARE)
+		dai_trigger(dd->dai, COMP_TRIGGER_STOP, dev->direction);
```

Comment:
can this also be entered when the pipeline is actually streaming? What would happen then?

Reply:
Looking at the CI results, this is not a proper solution :( 
If I remember correctly, component state is checked on ipc handling level before freeing. Although I'm not sure what may happen if an invalid reset is conducted (f.e. what happens if dma buffer is freed, but dma is not stopped?).